agent_build/docker/Dockerfile.base

```diff
@@ -0,0 +1,81 @@
+# Base image that creates all necessary dependencies for the scalyr-agent docker image.
+# NOTE: multi-stage builds require Docker 17.05 or greater
+
+# Suffix for the python dockerhub image. For now can be:
+#   - 'slim' for buster based image
+#   - 'alpine' for alpine based image.
+ARG BASE_IMAGE_SUFFIX
+
+# Install dependency packages for buster.
+FROM python:3.8.10-slim as scalyr-dependencies-slim
+MAINTAINER Scalyr Inc <support@scalyr.com>
+
+RUN apt-get update && apt-get install -y build-essential git tar curl
+
+
+# Install dependency packages for alpine.
+FROM python:3.8.10-alpine as scalyr-dependencies-alpine
+MAINTAINER Scalyr Inc <support@scalyr.com>
+
+RUN apk update && apk add --virtual build-dependencies \
+    binutils \
+    build-base \
+    gcc \
+    g++ \
+    make \
+    curl \
+    python3-dev \
+    patchelf \
+    git
+
+# Install rust and cargo. It is needed to build some of the Python dependencies of the agent.
+FROM scalyr-dependencies-$BASE_IMAGE_SUFFIX as scalyr-dependencies-rust-
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+RUN rustup toolchain install nightly
+RUN rustup default nightly
```

Comment:
I see you removed manylinux2014 wheel workaround, do you think we don't need it anymore?

With that workaround, we can re-use existing zstandard + orjson wheels for arm64 (yes, sadly just arm64 and we still need to build it from scratch for alpine + armv7).

Reply:
If we agree on the first slow build and then we rely on caching, then, I guess, this workaround won't win more time for us anymore.

Reply:
I guess it may still save us some time during the build of the first base image :)